### PR TITLE
fix fetchAccessTokenWithAssertionCredentials

### LINKED ISSUE #717: $client->refreshTokenWithAssertion() is giving error
$client->refreshTokenWithAssertion() is giving error.
it gives following error

PHP Fatal error:  Uncaught exception 'LogicException' with message 'config parameter "signing_key" must be set to refresh a token with assertion' in /Users/CFADMIN/Work/googleanalytics/google-api-php-client/src/Google/Client.php:201
Stack trace:
#0 /Users/CFADMIN/Work/googleanalytics/google-api-php-client/src/Google/Client.php(190): Google_Client->fetchAccessTokenWithAssertion()
#1 /Users/CFADMIN/Work/googleanalytics/HelloAnalytics.php(27): Google_Client->refreshTokenWithAssertion()
#2 /Users/CFADMIN/Work/googleanalytics/HelloAnalytics.php(101): getService()
#3 {main}

can you please tell me why is giving error. and how to resolve this.

### PR DESCRIPTION
**Pull request by bshaffer**
fixes #717

### CODE REVIEW

File: tests/Google/ClientTest.php [hunk 1]
```diff
@@ -386,4 +386,19 @@ public function testRefreshTokenSetsValues()
     $token = $client->getAccessToken();
     $this->assertEquals($token['id_token'], "ID_TOKEN");
   }
+
```
Comment (SurferJeffAtGoogle) [inline tests/Google/ClientTest.php]: Add a test that checks the Google_Client::setAuthConfig() code path.

**Comment by tmatsuo** [comment]
I'd appreciate it if you could add little bit more context. What problem this patch will fix? Is there an original bug?

**Comment by bshaffer** [comment]
@tmatsuo yes, see #717

File: tests/Google/ClientTest.php (hunk 1 above)
Comment (bshaffer) [inline tests/Google/ClientTest.php]: done!

**Comment by SurferJeffAtGoogle** [comment]
LGTM.

**Pull request merged**